### PR TITLE
review filters

### PR DESCRIPTION
**Pull request by eparejatobes**

### CODE REVIEW

**Comment by eparejatobes** [comment]
@rtobes @marina-manrique 

This is about detecting misassignments through comparing the initial assignment with the LCA mg7 result of running the ref db on itself.

Now, if a sequence has only one assignment, we are considering it valid, whatever the mg7 assignment is. Why? Is it on purpose?

**Comment by eparejatobes** [comment]
@laughedelic please write there in filter2 what your implementation is supposed to be doing.

**Comment by eparejatobes** [comment]
@laughedelic and check `TODO`s here

**Comment by laughedelic** [comment]
OK. So should I do it in #32?

**Comment by eparejatobes** [comment]
Better do it here, I want to merge this PR first and then do #32

File: src/main/scala/filter3.scala [hunk 1]
```diff
@@ -28,73 +28,78 @@ case object mg7results extends Bundle() {
...
-             that are too different from the LCA obtained from MG7,
-             i.e. are not descendants of its parent */
-          case Some(lcaParent) => {
-            val (acceptedTaxas, rejectedTaxas) = taxas.partition { taxa =>
-
-              taxonomyGraph.getNode(taxa).map { node =>
-                // lcaParent is in the lineage:
-                node.lineage.exists { _.id == lcaParent.id }
-              }.getOrElse(false)
+        // TODO why this is so?
```
Comment (laughedelic) [inline src/main/scala/filter3.scala]: See https://github.com/ohnosequences/db.rna16s/pull/32#discussion_r71972097

File: src/main/scala/filter3.scala [hunk 2]
```diff
@@ -28,73 +28,78 @@ case object mg7results extends Bundle() {
...
+        if (taxas.length == 1)
+          accept(id, taxas, fasta)
+        else {
+
+          id2mg7lca.get(id)
+            // TODO if the resulting ID is not in the bio4j taxonomy it should be discarded
```
Comment (laughedelic) [inline src/main/scala/filter3.scala]: It already happened in the previous filters.

File: src/main/scala/filter2.scala [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
 import era7bio.db._, csvUtils._, collectionUtils._
-
 import ohnosequences.fastarious.fasta._
 import ohnosequences.statika._
-
 import com.github.tototoshi.csv._
 import better.files._
 
-case object filter2 extends FilterDataFrom(filter1)() {
+// TODO I don't understand what this is trying to do, and how.
```
Comment (laughedelic) [inline src/main/scala/filter2.scala]: Does this [paragraph](https://github.com/ohnosequences/db.rna16s/blob/f37b457138f78d08c4941c46cb5738486c933179/readme.md#filter2) help?

File: src/main/scala/filter3.scala (hunk 1 above)
Comment (eparejatobes) [inline src/main/scala/filter3.scala]: OK, will wait for @rtobes and @marina-manrique then. I'm not quite happy with the current state :)

File: src/main/scala/filter3.scala (hunk 2 above)
Comment (eparejatobes) [inline src/main/scala/filter3.scala]: If so, should be documented.

File: src/main/scala/filter2.scala (hunk 3 above)
Comment (eparejatobes) [inline src/main/scala/filter2.scala]: not really. That's about _what_ this needs to do, but not _how_

File: src/main/scala/filter3.scala (hunk 2 above)
Comment (laughedelic) [inline src/main/scala/filter3.scala]: Here?

Comment (eparejatobes) [inline src/main/scala/filter3.scala]: Yes!

File: src/main/scala/filter2.scala (hunk 3 above)
Comment (laughedelic) [inline src/main/scala/filter2.scala]: OK. See 1bccc3f.

Comment (eparejatobes) [inline src/main/scala/filter2.scala]: I want first a readable global description of what `filterData` does. I'm merging this anyway, and we will do this in a separate PR.

Comment (laughedelic) [inline src/main/scala/filter2.scala]: It's a bit confusing. First I asked if you want the general description and you said 

> not really. That's about _what_ this needs to do, but not _how_

Now it's another way round..

Comment (eparejatobes) [inline src/main/scala/filter2.scala]: > Now it's another way round..

I don't think so; let me reword it a bit then: If you didn't wrote the code in https://github.com/ohnosequences/db.rna16s/commit/1bccc3f18ad08fee1aed82bcccc5be0df000507b, the comments you added doesn't help much in understanding how this is dropping redundant assignments; what I'm asking you to write is a general description of the implementation, not only comments for some vals inside a method.

Comment (laughedelic) [inline src/main/scala/filter2.scala]: This is more helpful
